ansible/roles/test/files/ptftests/sonic.py

```diff
@@ -0,0 +1,618 @@
+        self.do_cmd('exit')
+        self.do_cmd('exit')
+
+    def verify_bgp_neigh_state(self, dut=None, state="Active"):
+        bgp_state = {}
+        bgp_state['v4'] = bgp_state['v6'] = False
+        for cmd, ver in [('show ip bgp summary | json', 'v4'), ('show ipv6 bgp summary | json', 'v6')]:
+            output = self.do_cmd(cmd)
+            data = '\n'.join(output.split('\r\n')[1:-1])
+            obj = json.loads(data)
+
+            if state == 'down':
+                if 'vrfs' in obj:
+                    # return True when obj['vrfs'] is empty which is the case when the bgp state is 'down'
+                    bgp_state[ver] = not obj['vrfs']
+                else:
+                    self.fails.add('Verify BGP %s neighbor: Object missing in output' % ver)
+            else:
+                if 'vrfs' in obj and 'default' in obj['vrfs']:
+                    obj = obj['vrfs']['default']
+                    if 'peers' in obj:
+                        bgp_state[ver] = (obj['peers'][dut[ver]]['peerState'] in state)
+                    else:
+                        self.fails.add('Verify BGP %S neighbor: Peer attribute missing in output' % ver)
+                else:
+                    self.fails.add('Verify BGP %s neighbor: Object missing in output' % ver)
+        return self.fails, bgp_state
+
+    def change_neigh_lag_state(self, intf, is_up=True):
+        state = ['shut', 'no shut']
+        self.do_cmd('configure')
+        is_match = re.match('(Port-Channel|Ethernet)\d+', intf)
+        if is_match:
+            output = self.do_cmd('interface %s' % intf)
+            if 'Invalid' not in output:
+                self.do_cmd(state[is_up])
+                self.do_cmd('exit')
+        self.do_cmd('exit')
+
+    def change_neigh_intfs_state(self, intfs, is_up=True):
+        for intf in intfs:
+            self.change_neigh_lag_state(intf, is_up=is_up)
+
+    def verify_neigh_lag_state(self, lag, state="connected", pre_check=True):
+        states = state.split(',')
+        lag_state = False
+        msg_prefix = ['Postboot', 'Preboot']
+        is_match = re.match('(Port-Channel|Ethernet)\d+', lag)
+        if is_match:
+            output = self.do_cmd('show interfaces %s | json' % lag)
+            if 'Invalid' not in output:
+                data = '\n'.join(output.split('\r\n')[1:-1])
+                obj = json.loads(data)
+
+                if 'interfaces' in obj and lag in obj['interfaces']:
+                    lag_state = (obj['interfaces'][lag]['interfaceStatus'] in states)
+                else:
+                    self.fails.add('%s: Verify LAG %s: Object missing in output' % (msg_prefix[pre_check], lag))
+                return self.fails, lag_state
+
+        self.fails.add('%s: Invalid interface name' % msg_prefix[pre_check])
+        return self.fails, lag_state
+
+    def verify_neigh_lag_no_flap(self):
+        flap_cnt = sys.maxint
+        output = self.do_cmd('show interfaces Po1 | json')
```

Comment:
No, it won't. There's a fair number of things in this file that won't work for SONiC, but I've kept them for now mostly because either they're not being executed for regular warm reboot, or they don't happen to break anything in a bad way.